homeassistant/helpers/entity_registry.py

```diff
@@ -1113,6 +1113,9 @@ def async_device_modified(
                 ):
                     self.async_remove(entity.entity_id)
                 else:
+                    if entity.entity_id not in self.entities:
```

Comment:
While we're looping over the entities connected to the device, the branch above which removes an entity may cause another entity to be removed (seen when updating tests for helper entities).